### PR TITLE
API v4 update

### PR DESCRIPTION
**Pull request by nurfet-becirevic**

### CODE REVIEW

File: tests/test_image.py [hunk 1]
```diff
@@ -34,20 +35,45 @@ def test_get_image(self):
 
         self.assertEqual(image['type'], 'image')
         self.assertEqual(image['id'], self.image['id'])
+        assertRegex(self, image['id'], self.resource['uuid_match'])
+        self.assertGreater(len(image['properties']['name']), 0)
+        self.assertIsNone(image['properties']['description'])
+        self.assertGreater(image['properties']['size'], 0)
+        self.assertNotEquals(image['properties']['name'], "")
```
Comment (alibazlamit) [inline tests/test_image.py]: `test_image.py:42: DeprecationWarning: Please use assertNotEqual instead.
  self.assertNotEquals(image['properties']['name'], "")`

**Comment by nurfet-becirevic** [comment]
Yes we could pass an invalid data center id but we'll get different error
message, i.e., `Resource does not exist` instead of `Attribute 'location'
is required`. In GET negative test cases we're covering the error `Resource
does not exist`. @edevenport, what do you think?

File: tests/test_loadbalancer.py [hunk 2]
```diff
@@ -175,6 +174,30 @@ def test_get_balanced_nic(self):
 
         self.assertEqual(balanced_nic['id'], self.nic1['id'])
         self.assertEqual(balanced_nic['type'], 'nic')
+        self.assertEqual(balanced_nic['properties']['name'], self.nic1['properties']['name'])
+        self.assertEqual(balanced_nic['properties']['dhcp'], self.nic1['properties']['dhcp'])
+        self.assertIsInstance(balanced_nic['properties']['nat'], bool)
+        self.assertIsInstance(balanced_nic['properties']['firewallActive'], bool)
+        self.assertGreater(balanced_nic['properties']['ips'], 0)
```
Comment (alibazlamit) [inline tests/test_loadbalancer.py]:
```
test_loadbalancer.py", line 181, in test_get_balanced_nic
self.assertGreater(balanced_nic['properties']['ips'], 0)
File "C:\Users\Ali\AppData\Local\Programs\Python\Python36-32\lib\unittest\case.py", line 1210, in assertGreater
    if not a > b:
TypeError: '>' not supported between instances of 'list' and 'int'
```

File: tests/test_loadbalancer.py [hunk 3]
```diff
@@ -130,25 +123,31 @@ def test_update_loadbalancer(self):
         loadbalancer = self.client.update_loadbalancer(
             datacenter_id=self.datacenter['id'],
             loadbalancer_id=self.loadbalancer['id'],
-            name="updated name")
+            name=self.resource['loadbalancer']['name']+' - RENAME')
```
Comment (alibazlamit) [inline tests/test_loadbalancer.py]:
```
test_loadbalancer.py", line 126, in test_update_loadbalancer
 in _perform_request
    raise PBError(code, msg, url)
profitbricks.errors.PBError: (500, [{'errorCode': '301', 'message': 'Oops! Something went very wrong. Please contact the administrator'}], 'https://api.profitbricks.com/cloudapi/v4/datacenters/7fcee79c-6ac7-4fed-aa9d-4579e17b9ec5/loadbalancers/e8ef6ed9-cbf5-49fe-9b5d-c8801de79438')

```

Comment (nurfet-becirevic) [inline tests/test_loadbalancer.py]: This seems to be an intermittent failure.

**Pull request merged**